src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -636,11 +643,11 @@ enum class GeoFilterType { NormalGeos,
 void TaskSketcherElements::onListMultiFilterItemChanged(QListWidgetItem* item)
 {
     {
+        QSignalBlocker sigblk(filterList);
         int start = 4; //From 4 to the end, it's the geometry types (line, circle, arc...)
-        QSignalBlocker sigblk(ui->listMultiFilter);
-        if (item == ui->listMultiFilter->item(static_cast<int>(GeoFilterType::AllGeosTypes))) {
-            for (int i = start; i < ui->listMultiFilter->count(); i++) {
-                ui->listMultiFilter->item(i)->setCheckState(item->checkState());
+        if (item == filterList->item(static_cast<int>(GeoFilterType::AllGeosTypes))) {

```

Comment:
There's actually a minor off-by-one error hiding in here that will be exposed more clearly if we eliminate the magic number: what this really amounts to as currently written is
```
int start = static_cast<int>(GeoFilterType::AllGeosTypes);
```
which is not quite right, of course, because that's `item` itself! No need to set `item`'s check state to its own check state, after all. I'd tweak this a little bit:
```
int indexOfAllTypes = static_cast<int>(GeoFilterType::AllGeosTypes);
if (item == filterList->item(indexOfAllTypes ))) {
    for (int i = indexOfAllTypes +1; i < filterList->count(); i++) {
```
or something along those lines.

(This is also where you will want to add your code to handle setting the state to `Qt::PartiallyChecked`.)

Reply:
Ah yes start should be 5 since we added the 'internal' filter. I will fix it and add the PartiallyChecked.

Reply:
Actually for PartiallyChecked, I'm not sure it can be easily added to the constraint widget as well since it has a more complex filter. And I prefer if they both look the same. So maybe we can keep as is for now and when constraint PR merge too we can see if it can be added to both at the same time.

Reply:
I'd argue this isn't really a "preference" issue: `Qt::PartiallyChecked` is literally intended for exactly this use case, and is the correct state of the "parent" widget. I'm OK with this being done in a separate PR, but it definitely should be done.

Reply:
I pushed a commit to add Qt::PartiallyChecked. It seems to be working as expected. Please check if it's done correctly and work as you expected.

Reply:
> I pushed a commit to add Qt::PartiallyChecked. It seems to be working as expected. Please check if it's done correctly and work as you expected.

Just a sidenote. Especially when fixing, you should take the habit to make separate commits for separate points. Because then it will be very hard to squash. Also using `git commit --fixup SHA` is of great help.
We'll see later how to get a clean history now. ;)